portal/models/reference.py

```diff
@@ -72,6 +75,29 @@ def parse(cls, reference_dict):
                         reference_text))
                 return result
 
+        match = re.compile('[Oo]rganization/(.+)/(.+)').search(reference_text)
+        if match:
+            try:
+                id_system = match.groups()[0]
+                id_value = match.groups()[1]
+            except:
+                raise ValueError('Identifier values not found in ' \
+                    'reference {}'.format(reference_text))
+            with db.session.no_autoflush:
+                result = Organization.query.join(
+                      OrganizationIdentifier).join(Identifier).filter(and_(
+                          Organization.id==OrganizationIdentifier.organization_id,
+                          OrganizationIdentifier.identifier_id==Identifier.id,
+                          Identifier.system==id_system,
+                          Identifier._value==id_value))
+            if not result.count():
+                raise MissingReference("Reference not found: {}".format(
+                    reference_text))
+            elif result.count() > 1:
```

Comment:
I think for consistency, as well as the potential need to catch it in view code somewhere, we should define our own, say MultipleReference exception.